### PR TITLE
Updated contribution documentation.

### PR DESCRIPTION
**Pull request by evildmp**
* Added notes about 3.4 LTS version.
* Moved development documentation to a new page.
* Corrected commit-squashing example in the documentation.

### CODE REVIEW

File: docs/contributing/development-policies.rst [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+The `django CMS developers email list <http://groups.google.com/group/django-cms-developers>`_ serves as gathering
+point for developers. We submit ideas and proposals prior to the roadmap goals.
+
+django CMS 3.4 will be the first "LTS" ("Long-Term Support") release of the application. *Long-term support* means that
+this version will continue to receive security and other critical updates for 24 months after its first release.
+
+Any updates it does receive will be backward-compatible and will not alter functional behaviour. This means that users
+can deploy this version confident that keeping it up-to-date requires only easily-applied security and other critical
+updates, until the next LTS release.
```
Comment (mkoistinen) [inline docs/contributing/development-policies.rst]: I think this should be phrased "It is *our aim* that any updates... but we reserve the right to push incompatible changes if required to preserve security."

File: docs/contributing/development-policies.rst [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    85d925c Updated commit policy based on feedback.
+
+The bottom three commits are just noise. They don't represent development of the code base. The four commits
+should be squashed into a single, meaningful, commit::
+
+    85d925c Updated contribution policies.
+
+
+How to squash commits
+---------------------
```
Comment (mkoistinen) [inline docs/contributing/development-policies.rst]: Perhaps we could state that "GitHub does this for us, but if you prefer to do it manually..."

File: docs/contributing/management.rst [hunk 3]
```diff
@@ -31,7 +31,7 @@ Raising an issue
         or in any other public forum until we have had a chance to deal with it.
 
 Except in the case of security matters, of course, you're welcome to raise issues in any way that
-suits you - :ref:`on one of our email lists, or the IRC channel <community-resources>` or in person
+suits you - :ref:`on one of our email lists, or the IRC channel <development-community>` or in person
```
Comment (mkoistinen) [inline docs/contributing/management.rst]: What is this exactly? I'm not familiar with a IRC channel "development-community". Does this resolve into '#django-cms" somehow?

File: docs/reference/cli.rst [hunk 4]
```diff
@@ -57,21 +60,22 @@ Identifies and deletes orphaned plugins.
...
+* have a ``plugin_type`` that is no longer even installed
 * have no corresponding saved instance in that particular plugin type's table
 
 Such plugins will cause problems when trying to use operations that need to copy
 pages (and therefore plugins), which includes ``cms moderator on`` as well as page
 copy operations in the admin.
 
-It is advised to run ``cms list plugins`` periodically, and ``cms
+It is recommended to run ``cms list plugins`` periodically, and ``cms
 delete-orphaned-plugins`` when required.
```
Comment (mkoistinen) [inline docs/reference/cli.rst]: Perhaps mention "...after you've taken a backup of your database."

**Comment by mkoistinen** [comment]
Made several notes, but looks really great!

File: docs/contributing/management.rst (hunk 3 above)
Comment (evildmp) [inline docs/contributing/management.rst]: It's a link to the "django CMS's development community" heading in ``ref/development-community``.

File: docs/reference/cli.rst (hunk 4 above)
Comment (evildmp) [inline docs/reference/cli.rst]: Will do.

Comment (evildmp) [inline docs/reference/cli.rst]: Oh, it already does above!

**Pull request merged**